ui/lib/core/addon/components/ttl-picker2.js

```diff
@@ -48,24 +48,29 @@ const goSafeConvertFromSeconds = (seconds, unit) => {
   return convertFromSeconds(seconds, u) + u;
 };
 
-export default Component.extend({
-  layout,
-  enableTTL: false,
-  label: 'Time to live (TTL)',
-  helperTextDisabled: 'Allow tokens to be used indefinitely',
-  helperTextEnabled: 'Disable the use of the token after',
-  description: '',
-  time: '', // if defaultValue is NOT set, then do not display a defaultValue.
-  unit: 's',
-  initialValue: null,
-  changeOnInit: false,
-  hideToggle: false,
-
-  init() {
-    this._super(...arguments);
-    const value = this.initialValue;
-    const enable = this.initialEnabled;
-    const changeOnInit = this.changeOnInit;
+export default class TtlPicker2Component extends Component {
+  @tracked enableTTL = false;
+  @tracked time = ''; // if defaultValue is NOT set, then do not display a defaultValue.
+  @tracked unit = 's';
+
+  get label() {
+    return this.args.label || 'Time to live (TTL)';
+  }
+  get helperTextDisabled() {
+    return this.args.helperTextDisabled || 'Allow tokens to be used indefinitely';
+  }
+  get helperTextEnabled() {
+    return this.args.helperTextEnabled || 'Disable the use of the token after';
+  }
+  // initialValue: null,
+  // changeOnInit: false,
+  // hideToggle: false,
```

Comment:
> I see what you mean with the template being less than ideal when incorporating the toggle but I think it's still worth it to get rid of the form component. Thanks for doing that! It's too bad that the `Toggle` component has to wrap everything. I thought that it would be standalone similar to a checkbox.

Yeah, it doesn't *have* to wrap everything, but in order to get behavior where you can click on the label to toggle the content, it does have to wrap. A future update could be something where the parent can override the input ID so that we can match on a custom label, but I think that's out of scope for now.